build.gradle

```
@@ -50,6 +50,11 @@ subprojects {
       showStackTraces true
       showCauses true
     }
+
+    if (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_21)) {
```

```suggestion
    if (javaVersion.isCompatibleWith(JavaVersion.VERSION_21)) {
```
I think you want this variant:
 * Test (receiver here).javaVersion is the resolved tool chain for this task
 * JavaVersion.current() is the Gradle wrapper daemon JVM. Those two could differ.

---

For multiplatform, doing `Test.javaVersion` throws this error

```
tion: The value for task ':test-projects:multiplatform-plugin-with-android:testDebugUnitTest' property 'javaLauncher' is final and cannot be changed any further.
```